src/main/java/it/gov/pagopa/common/web/exception/ServiceExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package it.gov.pagopa.common.web.exception;
+
+
+import it.gov.pagopa.reward.exception.ErrorManagerExtended;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.server.ServerWebExchange;
+
+import java.util.Map;
+
+@RestControllerAdvice
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class ServiceExceptionHandler {
+    private final ErrorManager errorManager;
+    private final ErrorManagerExtended errorManagerExtended;
+    private final Map<Class<? extends ServiceException>, HttpStatus> transcodeMap;
+
+    public ServiceExceptionHandler(ErrorManager errorManager, ErrorManagerExtended errorManagerExtended, Map<Class<? extends ServiceException>, HttpStatus> transcodeMap) {
+        this.errorManager = errorManager;
+        this.errorManagerExtended = errorManagerExtended;
+        this.transcodeMap = transcodeMap;
+    }
+
+    /**
+     * Return two types of ResponseEntity:
+     * <ol>
+     *     <li>ErrorDTO: when ServiceException not contains a specific response</li>
+     *     <li>SynchronousTransactionResponseDTO: when ServiceException contains a specific response</li>
+     * </ol>
+     */
+    @ExceptionHandler(ServiceException.class)
+    protected ResponseEntity<?> handleException(ServiceException error, ServerWebExchange exchange) {
+        if(null != error.getResponse()){
+            return errorManagerExtended.synchronousTrxHandleException(error,transcodeException(error));
```

Comment:
A "common" class should not have dependencies towards not "common" class.
Let's resolve the dependency moving here the code (the "reward" class could be entirely removed this way)